src/next-vt.tsx

```diff
@@ -1,56 +1,64 @@
 import {
-    useViewTransition
+    useViewTransition,
+    SuspendViewTransition
 } from "./react-vt";
-
-import {
+import React, {
     useEffect,
-    Suspense
+    Suspense,
+    FC
 } from "react";
 
 export { SuspendViewTransition, useViewTransition } from "./react-vt";
 
+interface UseNextRouterViewTransitionsProps {
+    events: {
+        on: (event: string, handler: () => void) => void;
+        off: (event: string, handler: () => void) => void;
+    };
+}
+
 /**
  * Performs CSS view transitions automatically when a NextJS navigation takes place.
  *
  * Use this hook in your _app.js.
  *
  */
 export function useNextRouterViewTransitions({
-    events
-}) {
+                                                 events
+                                             }: UseNextRouterViewTransitionsProps): void {
     const {
         startViewTransition,
         suspendViewTransitionCapture,
         resumeViewTransitionCapture
     } = useViewTransition();
 
     useEffect(() => {
-        function beginNavigation() {
+        function beginNavigation(): void {
             startViewTransition();
             suspendViewTransitionCapture();
         };
 
-        function endNavigation() {
+        function endNavigation(): void {
             resumeViewTransitionCapture();
         }
 
         events.on("routeChangeStart", beginNavigation);
         events.on("routeChangeComplete", endNavigation);
         return () => {
-            events.off("routerChangeStart", beginNavigation);
+            events.off("routeChangeStart", beginNavigation);
             events.off("routeChangeComplete", endNavigation);
         };
     }, []);
 }
 
-function RouterEventsNotifier() {
-  usePathname();
-  useSearchParams();
-  return null;
+function RouterEventsNotifier(): null {
+    // usePathname();
```

Comment:
If you want to erase it erase rather than comment out.

Reply:
Okey, please check last patch